sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnReflector.java

```diff
@@ -499,7 +500,18 @@ private boolean usesContext(AdditionalParameter param) {
      */
     private Constructor<? extends DoFnInvoker<?, ?>> createInvokerConstructor(
         @SuppressWarnings("rawtypes") Class<? extends DoFnWithContext> clazz) {
+
+      final TypeDescription clazzDescription = new TypeDescription.ForLoadedType(clazz);
+
       DynamicType.Builder<?> builder = new ByteBuddy()
+          // Create subclasses inside the target class's package
+          .with(new NamingStrategy.SuffixingRandom("auxiliary") {
+                @Override
+                public String subclass(TypeDescription.Generic superClass) {
+                  return super.name(clazzDescription);
```

Comment:
If I'm reading this right, it actually creates the subclass inside the target class, correct? That seems reasonable, since it ensures access to private types/methods as well.

That said, it is likely reasonable to say that the annotated methods should not be private (the DoFn can be private, but the methods should be public), to mimic the "cannot reduce visibility" behavior we had with `@Override`.